src/WebAsmReader.h

```diff
@@ -0,0 +1,124 @@
+#ifndef WEBASMREADER_H
```

Comment:
Or actually, how about a `Reader` helper class for reading, and `Writer` class for writing. Also no need for `Wasm` as a prefix as it's in the namespace already.